pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,1484 @@
+            for vl in setNumbers:
+                groupInfo['SET' + vl] = {
+                    'setObject': setInfo['SET' + vl]['setObject'],
+                    'indices': [],
+                }
+
+            for gp in groupSetPairs:
+                if gp[0] not in groupInfo['SET' + gp[1]]['indices']:
+                    groupInfo['SET' + gp[1]]['indices'].append(gp[0])
+
+            indexCounter = 0
+            for ky, vl in groupInfo.items():
+                if vl['setObject'] in ComponentSet(latex_component_map.keys()):
+                    indexNames = latex_component_map[vl['setObject']][1]
+                    if len(indexNames) != 0:
+                        if len(indexNames) < len(vl['indices']):
+                            raise ValueError(
+                                'Insufficient number of indices provided to the overwrite dictionary for set %s'
+                                % (vl['setObject'].name)
+                            )
+                        for i in range(0, len(vl['indices'])):
+                            ln = ln.replace(
+                                '__I_PLACEHOLDER_8675309_GROUP_%s_%s__'
+                                % (vl['indices'][i], ky),
+                                indexNames[i],
+                            )
+                    else:
+                        for i in range(0, len(vl['indices'])):
+                            ln = ln.replace(
+                                '__I_PLACEHOLDER_8675309_GROUP_%s_%s__'
+                                % (vl['indices'][i], ky),
+                                alphabetStringGenerator(indexCounter, True),
+                            )
+                            indexCounter += 1
+                else:
+                    for i in range(0, len(vl['indices'])):
+                        ln = ln.replace(
+                            '__I_PLACEHOLDER_8675309_GROUP_%s_%s__'
+                            % (vl['indices'][i], ky),
+                            alphabetStringGenerator(indexCounter, True),
+                        )
+                        indexCounter += 1
+
+        latexLines[jj] = ln
+
+    pstr = '\n'.join(latexLines)
+
+    vrIdx = 0
+    new_variableMap = ComponentMap()
+    for i in range(0, len(variableList)):
+        vr = variableList[i]
+        vrIdx += 1
+        if isinstance(vr, ScalarVar):
+            new_variableMap[vr] = vr.name
+        elif isinstance(vr, IndexedVar):
+            new_variableMap[vr] = vr.name
+            for sd in vr.index_set().data():
+                # vrIdx += 1
+                sdString = str(sd)
+                if sdString[0] == '(':
+                    sdString = sdString[1:]
+                if sdString[-1] == ')':
+                    sdString = sdString[0:-1]
+                new_variableMap[vr[sd]] = vr[sd].name
+        else:
+            raise DeveloperError(
+                'Variable is not a variable.  Should not happen.  Contact developers'
+            )
```

Comment:
Is this code reachable? You would have already hit this the first time you enumerated through variableList, right?

Reply:
you are right, but I prefer to leave it in.  Only two things are infinite, the universe and the capacity for users to do things I don't expect.  And I'm not sure about the first